scipy/stats/kde.py

```diff
@@ -569,7 +569,13 @@ def _compute_covariance(self):
 
         self.covariance = self._data_covariance * self.factor**2
         self.inv_cov = self._data_inv_cov / self.factor**2
-        self._norm_factor = sqrt(linalg.det(2*pi*self.covariance))
+        itemsize = self.covariance.dtype.itemsize
+        if itemsize in (12, 16):
+            self.log_det = np.abs(np.log(linalg.det(2.0*pi*self.covariance)))
+            self.log_sign = np.log(np.sign(self.log_det))
+        else:
+            sign, self.log_det = np.linalg.slogdet(2.0*pi*self.covariance)
+            self.log_sign = np.log(sign)
```

Comment:
For the linear algebra shims, @ilayn or @pmla come to mind for a quick check (not my field of expertise)

Reply:
To avoid the typical overflow to inf or underflow to zero, log-det needs to be computed via first getting the cholesky factor of covariance matrix (assuming always pos-def) and then multiplying the diagonal elements of the factor and then taking the log (which is a sum by the log product rule) so:  

```python
L = cholesky(self.covariance*2*pi)
self.log_det = 2*np.log(np.diag(L)).sum()
```

Reply:
Awesome, this is a point that I had overlooked, and it makes the code simpler with the same functionality. Added in https://github.com/scipy/scipy/pull/12989/commits/9af9cf3df18fa2e395f9c6b68228eb7468bac9fd.